spec/lib/post_handler_spec.rb

```diff
@@ -92,6 +93,60 @@
         }
       end
     end
+
+    describe "when changed ownership of the post" do
+      describe "from category expert to another expert" do
+        it "deletes the post approved group name" do
+          post = create_post(topic_id: topic.id, user: expert)
+          CategoryExperts::PostHandler.new(post: post).mark_post_as_approved
+
+          expect(post.custom_fields[CategoryExperts::POST_APPROVED_GROUP_NAME]).to eq(group.name)
+
+          PostOwnerChanger.new(
+            post_ids: [post.id],
+            topic_id: topic.id,
+            new_owner: second_expert,
+            acting_user: admin,
+          ).change_owner!
+
+          expect(post.reload.custom_fields[CategoryExperts::POST_APPROVED_GROUP_NAME]).to eq(nil)
+        end
+      end
+
+      describe "from category expert to not an expert" do
+        it "deletes the post approved group name" do
+          post = create_post(topic_id: topic.id, user: expert)
+          CategoryExperts::PostHandler.new(post: post).mark_post_as_approved
+
+          expect(post.custom_fields[CategoryExperts::POST_APPROVED_GROUP_NAME]).to eq(group.name)
+
+          PostOwnerChanger.new(
+            post_ids: [post.id],
+            topic_id: topic.id,
+            new_owner: user,
+            acting_user: admin,
+          ).change_owner!
+
+          expect(post.reload.custom_fields[CategoryExperts::POST_APPROVED_GROUP_NAME]).to eq(nil)
```

Comment:
Yeah this one makes sense, as you're transfering it to not an expert. Maybe in the spec above, the user isn't actually an expert for the category or something?